contracts/minters/open-edition-minter/src/contract.rs

```diff
@@ -264,12 +266,93 @@ pub fn execute_mint_to(
     _execute_mint(deps, env, info, action, true, Some(recipient))
 }
 
+fn _pay_mint_if_not_contract(
+    this_contract: Addr,
+    info: MessageInfo,
+    mint_price_with_discounts: Coin,
+    config_denom: String,
+) -> Result<Uint128, ContractError> {
+    let this_contract = this_contract.to_string();
+    match info.sender == this_contract {
+        true => Ok(Uint128::new(0)),
+        false => {
+            let payment = may_pay(&info, &config_denom)?;
+            if payment != mint_price_with_discounts.amount {
+                return Err(ContractError::IncorrectPaymentAmount(
+                    coin(payment.u128(), &config_denom),
+                    mint_price_with_discounts,
+                ));
+            }
+            Ok(payment)
+        }
+    }
+}
+
+fn _fairburn_if_not_contract_sender(
+    deps: &DepsMut,
+    this_contract: Addr,
+    info: MessageInfo,
+    mint_price_with_discounts: Coin,
+    mint_fee: Decimal,
+    factory_params: MinterParams<open_edition_factory::state::ParamsExtension>,
+) -> Result<(Response, Uint128), ContractError> {
+    let mut res = Response::new();
+    match info.clone().sender == this_contract {
+        true => Ok((res, Uint128::new(0))),
```

Comment:
It has to be, because when you send an NFT as payment, you are not also sending token. So what would you be fair burning? You can't fair burn an NFT.